fix #50561: start note entry on current page if no selection

MarcSabatella opened the pull request:
We have to start somewhere, and we have always chosen measure 1, but recently it has been suggested it would be better to start ont eh current page.  My code here starts entry at the first measure of the current page, so for a one page score (eg, a newly created score), it always starts in measure 1 just as it always has.  It's really just about losing one's place if one accidentally enters note input mode without selecting something first.

The code is not as trivial as I had hoped, but it is very well localized.  Perhaps there is a different chocie we could make that would be simpler but still effective, but I thought I would at least offer this.

lasconic commented on mscore/scoreview.cpp:
```diff
@@ -3167,6 +3167,29 @@ void ScoreView::startNoteEntry()
+      if (_score->selection().isNone()) {
+            // select first chordrest in first measure on first staff of current page
+            QPointF center = toLogical(QPoint(width() * 0.5, height() * 0.5));
+            Page* p = point2page(center);
```
That will probably fail in Continuous View (LayoutMode::Line). There is a single page in this case. So what about finding the first visible measure in the current view. Or the measure closer from the center of the current view?

Also point2page can be 0 in some extreme cases, we could try in the 4th corners if it fails in the center?

MarcSabatella replied:
I am not quite sure how to find first visible measure, but if I figure that out, it might make sense to do that for any view.  I used the first measure of page rather than measure closest to center to make sure we don't change current behavior in case of brand new empty score, where note entry always starts in first measure of score even if nothing selected.

MarcSabatella commented:
Updated.  Improved detection of page - try to find page in upper left quarant, then try all four corners of current view looking for hit.  One page is found, select the top left chordrest in view on that page.

Also fixes a crash that occured if you tried to enter note input mode with no measures in the score.  And speaking of crashes, the code I added is a bit more verbose than necessary in the name of "defensive coding".  I tried to bend over backwards to minimize any possibiltiy of crash on unexpected conditions.
